grpc.go

```diff
@@ -603,9 +607,30 @@ func (a *agentGRPC) CreateContainer(ctx context.Context, req *pb.CreateContainer
 		return emptyResp, err
 	}
 
+	if err := updateSharedPidNs(a.sandbox, ctr); err != nil {
+		return emptyResp, err
+	}
+
 	return emptyResp, a.postExecProcess(ctr, ctr.initProcess)
 }
 
+func updateSharedPidNs(s *sandbox, ctr *container) error {
+	// Populate the shared pid path only if this is an infra container and the
+	// shared path is empty. This means SandboxPidns has not been
+	// passed in the CreateSandbox request and a separate pause process has not been
+	// created. We treat the first container created as the infra container in that case
+	// and use its pid namespace in case it needs to be shared.
+	if s.sharedPidNs.path == "" && len(s.containers) == 1 {
```

Comment:
Please use `a.sandbox.sandboxPidNs` to do the checking, and clear `s.sharedPidNs.path` when removing the last container from sandbox. Then you don't need an empty `s.sharedPidNs.path` to indicate `a.sandbox.sandboxPidNs` being false and the check works when removing the last container from sandbox.

Reply:
I dont follow. Let me explain what I have done here.
If `a.sandbox.sandboxPidNs`, then we create our own pause process and  use that as the `s.sharedPidNs.path`. This is to satisfy the frakti scenario.
Usually either `a.sandbox.sandboxPidNs` will be set(frakti) or `ctr.useSandboxPidNs`(CRI).

If `a.sandbox.sandboxPidNs` is set then we continue using agent's pause process pidns as the shared pidns. If this is not set, and if `ctr.useSandboxPidNs` is set, then the first process is treated as the infra container and its pidns is shared. Because in this case a separate pause process has not been created.
Thats why I check for the blank value for `s.sharedPidNs.path`. It is populated once either by the pause process that agent creates or by the very first container(infra container, which is the last one to exit)

Reply:
Also my runtime PR just sets `ctr.useSandboxPidNs`  in the cri-o and cri-containerd use cases. The flag `a.sandbox.sandboxPidNs` which existed previously is not set at all, I assume this will be set when called through frakti.

Reply:
@bergwolf When pid namespace is shared, this is very much true for k8s. The first container will be the last one to exit, since it is the pid 1 process of the sandbox. I have verified this for both cri-containerd and crio. If the pid 1 exits, the other containers will be terminated. 
For frakti which does not really on an infra container, this will be handled by creating a pause process by the agent by setting `a.sandbox.sharedPidNs` to true and in that case `s.sharedPidNs.path` will be populated with the pause process pidns and not that of any container.

Reply:
@amshinde The kata agent should be self-contained and not depend on its users to be correct internally. The kata agent is robust on its own, not JustWork(R) with k8s/docker/frakti. And the sandbox life cycle and container life cycle is different by design. That's why we have CreateSandbox and DestroySandbox APIs.

Reply:
@bergwolf Fair enough. Have fixed this.